# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/PDMFactory.h
# Commit: e3f920da2232
# Full Hash: e3f920da22320921473f385cb060ed7978d67df9
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P3: Only create PDMs that are supported in the current process. r=kamidphish
#   
#   Depends on D52797
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54876
# ==============================================================================

diff -r 44471f99280c -r e3f920da2232 dom/media/platforms/PDMFactory.h
--- a/dom/media/platforms/PDMFactory.h	Tue Oct 20 23:23:58 2020 +0000
+++ b/dom/media/platforms/PDMFactory.h	Tue Oct 20 23:24:27 2020 +0000
@@ -51,11 +51,21 @@
   static constexpr int kYUV444 = 3;
 
  private:
-  virtual ~PDMFactory();
+  virtual ~PDMFactory() = default;
+
   void CreatePDMs();
   void CreateNullPDM();
+  void CreateGpuPDMs();
+  void CreateRddPDMs();
+  void CreateDefaultPDMs();
+
+  template <typename DECODER_MODULE, typename... ARGS>
+  bool CreateAndStartupPDM(ARGS&&... aArgs) {
+    return StartupPDM(DECODER_MODULE::Create(std::forward<ARGS>(aArgs)...));
+  }
   // Startup the provided PDM and add it to our list if successful.
-  bool StartupPDM(PlatformDecoderModule* aPDM, bool aInsertAtBeginning = false);
+  bool StartupPDM(already_AddRefed<PlatformDecoderModule> aPDM,
+                  bool aInsertAtBeginning = false);
   // Returns the first PDM in our list supporting the mimetype.
   already_AddRefed<PlatformDecoderModule> GetDecoder(
       const TrackInfo& aTrackInfo,